OSDOCS#10130: Doc support for ARM control plane with x86 compute nodes (#76246)
openshift/openshift-docs: modules/multi-architecture-modify-machine-set-aws.adoc

:_mod-docs-content-type: PROCEDURE
[id="multi-architecture-modify-machine-set-aws_{context}"]
= Adding a multi-architecture compute machine set to your AWS cluster

> To configure an AWS cluster with multi-architecture compute machines, you must create an AWS 64-bit ARM or 64-bit x86 compute machine set.

Reviewer: This incipit does not give a so meaningful information now that the secondary architecture can be any. Perhaps, something like 
> When your cluster relies on the multi-architecture payload, you can add nodes with diverse architectures. For example, you can add amd64 workers alongside other arm64 workers in a cluster using arm64 control plane machines. In this case, we define the secondary architecture as `amd64`.

Author: Updated as follows:

"After creating a multi-architecture cluster, you can add nodes with different architectures. For example, you can add 64-bit ARM compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture."

> To configure an AWS cluster with multi-architecture compute machines, you must create an AWS ARM64 or AMD64 compute machine set. This adds ARM64 or AMD64 compute nodes to your cluster.

Reviewer 2: Instead of **AMD64** and **ARM64**, we prefer to use **64-Bit ARM** and **64-bit x86**

Reviewer 2: also, you can remove: `This adds ARM64 or AMD64 compute nodes to your cluster`

Author: Updated as per the suggestion.

> After creating a multi-architecture cluster, you can add nodes with different architectures. For example, you can add 64-bit ARM compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.

Reviewer: ```suggestion
After creating a multi-architecture cluster, you can add nodes with different architectures. For example, you can add 64-bit ARM compute machines to a cluster that uses 64-bit AMD control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
```

Author: Updated as per the suggestion.

:_mod-docs-content-type: PROCEDURE
[id="multi-architecture-modify-machine-set-aws_{context}"]
= Adding a multi-architecture compute machine set to your AWS cluster

After creating a multi-architecture cluster, you can add nodes with different architectures.

> Following are the examples of adding multi-architecture compute machines to a multi-architecture cluster:

Reviewer 3: ```suggestion
You can add the following multi-architecture compute machines to a multi-architecture cluster:

*  Adding 64-bit x86 compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit ARM compute machines. In this case, 64-bit x86 is considered the secondary architecture.
* 64-bit ARM compute machines. You can add 64-bit ARM compute machines  to a cluster that uses 64-bit x86 control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
* 64-bit ARM compute machines. You can add 64-bit ARM compute machines to a cluster that uses 64-bit x86 control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.
```

Author: Updated as follows:

"You can add multi-architecture compute machines to a multi-architecture cluster in the following ways:"

* Adding 64-bit x86 compute machines to a cluster that uses 64-bit ARM control plane machines and already includes 64-bit ARM compute machines. In this case, 64-bit x86 is considered the secondary architecture.
* Adding 64-bit ARM compute machines to a cluster that uses 64-bit x86 control plane machines and already includes 64-bit x86 compute machines. In this case, 64-bit ARM is considered the secondary architecture.

include::snippets/about-multiarch-tuning-operator.adoc[]

.Prerequisites

* You installed the {oc-first}.
* You used the installation program to create an 64-bit ARM or 64-bit x86 single-architecture AWS cluster with the multi-architecture installer binary.

.Procedure

. Log in to the {oc-first}.

. Create a YAML file, and add the configuration to create a compute machine set to control the 64-bit ARM or 64-bit x86 compute nodes in your cluster.
+
--

> .Example `MachineSet` object for an AWS 64-bit ARM or x86 compute node:

Reviewer 3: Don't need the colon here.
```suggestion
.Example `MachineSet` object for an AWS 64-bit ARM or x86 compute node
```

          userDataSecret:
            name: worker-user-data
----
<1> Specify the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. If you have the {oc-first} installed, you can obtain the infrastructure ID by running the following command:
+
[source,terminal]
----
$ oc get -o jsonpath=‘{.status.infrastructureName}{“\n”}’ infrastructure cluster
----
<2> Specify the infrastructure ID, role node label, and zone.
<3> Specify the role node label to add.

> <4> Specify a 64-bit ARM supported Red Hat Enterprise Linux CoreOS (RHCOS) Amazon Machine Image (AMI) for your AWS zone for your OpenShift Container Platform nodes.

Author: Updated as follows:

"Specify a Red Hat Enterprise Linux CoreOS (RHCOS) Amazon Machine Image (AMI) for your AWS zone for your {product-title} nodes. The RHCOS AMI must be compatible with the machine architecture."

	  -o jsonpath='{.data.stream}' | jq \
	  -r '.architectures.<arch>.images.aws.regions."<region>".image'
----

> <5> Specify a 64-bit ARM supported machine type. For more information, see "Tested instance types for AWS 64-bit ARM"

Reviewer: ```suggestion
<5> Specify a machine type (with the same architecture as the AMI). For more information, see "Tested instance types for AWS 64-bit ARM"
```

Not sure how to describe this.

Author: Updated as per the suggestion.

<6> Specify the zone. For example, `us-east-1a`. Ensure that the zone you select offers 64-bit ARM machines.
<7> Specify the region. For example, `us-east-1`. Ensure that the zone you select offers 64-bit ARM machines.
--

. Create the compute machine set by running the following command:
+
[source,terminal]
----
$ oc create -f <file_name> <1>
----
<1> Replace `<file_name>` with the name of the YAML file with compute machine set configuration. For example: `aws-arm64-machine-set-0.yaml`.

.Verification

. View the list of compute machine sets by running the following command:
+
[source,terminal]
----
$ oc get machineset -n openshift-machine-api
----
+

> The output must include the AWS 64-bit ARM machine set that you created.

Reviewer: ```suggestion
The output must include the machine set that you created.
```

Author: Updated as per the suggestion.

	  -o jsonpath='{.data.stream}' | jq \
	  -r '.architectures.<arch>.images.aws.regions."<region>".image'
----
<5> Specify a machine type that aligns with the CPU architecture of the chosen AMI. For more information, see "Tested instance types for AWS 64-bit ARM"
<6> Specify the zone. For example, `us-east-1a`. Ensure that the zone you select has machines with the required architecture.
<7> Specify the region. For example, `us-east-1`. Ensure that the zone you select has machines with the required architecture.
--

. Create the compute machine set by running the following command:
+
[source,terminal]
----
$ oc create -f <file_name> <1>
----

> <1> Replace `<file_name>` with the name of the YAML file with compute machine set configuration. For example: `aws-arm64-machine-set-0.yaml`, or `aws-amd64-machine-set-0.yaml`

Reviewer 4: I would add a period at the end of the second sentence:
```suggestion
<1> Replace `<file_name>` with the name of the YAML file with compute machine set configuration. For example: `aws-arm64-machine-set-0.yaml`, or `aws-amd64-machine-set-0.yaml`.
```

Author: Updated as per the suggestion.
